### ISSUE TITLE
Error Extraño en Efecto de Spit At Opponent

### ISSUE THREAD

**Issue by piacabrera** [RodrigoToroIcarte/IIC2113-2023-1#208]
Hola! por alguna razón, en el test 12 de EffectsPart1 me pasa que al imprimir las cartas que puedo jugar, estas se imprimen correctamente pero por alguna razón, el efecto de Spit At Opponent se imprime mal. Dejo el ejemplo abajo:

(lo que imprimo yo)
1- [ACTION] *Spit At Opponent*. Info: 4F/0D, Action, Heel. Effect: You discard 1 card. Your opponent discards 4 cards.

(lo que debería imprimir)
1- [ACTION] *Spit At Opponent*. Info: 4F/0D, Action, Heel. Effect: Can only be played when you have 2 or more cards in your hand. Discard 1 card and then your opponent discards 4 cards.

La verdad trate de encontrar el error en el json, pero ahí el efecto esta bien y no se de donde mi programa saca el efecto que esta imprimiendo ahora, ya que nunca manipulo strings ni tengo escrito ese texto en mi codigo.

**Comment by RodrigoToroIcarte** [comment]
Es porque el json que subí en el código base estaba malo :(

Mandé un anuncio apenas nos dimos cuenta:

![Screenshot from 2023-10-24 16-19-01](https://github.com/RodrigoToroIcarte/IIC2113-2023-1/assets/7000518/6e039787-6bae-47ce-9a7d-6bc5ed6bd879)

El nuevo card.json se encuentra en canvas:
![Screenshot from 2023-10-24 16-19-39](https://github.com/RodrigoToroIcarte/IIC2113-2023-1/assets/7000518/0ba5ebf7-330c-4677-9c35-5b369034ea8f)

... dime si te funciona después del cambio.

**Comment by piacabrera** [comment]
Es que si cambie el json, y acabo de volver a hacerlo pero tengo el mismo error. :(
De hecho, en el Json el efecto es el largo, solo que por alguna razón al imprimirlo cuando se imprime la carta se pone el efecto corto

**Comment by RodrigoToroIcarte** [comment]
Me tincó que esto podría pasar.

Tienes que hacer click derecho en el cards.json. Luego ir a properties y ahí seleccionar en "Copy to output directory" la opción "Copy always", tal como sale aquí: https://github.com/RodrigoToroIcarte/IIC2113-2023-1/issues/27 (pero para el cards.json). Aceptas, vuelve a compilar, y debería funcionar.

Avísame si te funciona.

**Comment by piacabrera** [comment]
Oka, estoy siguiendo el tutorial de esa issue pero surge una duda, dentro de la carpeta de net6.0 tengo demasiados archivos ademas de data, como 47_Undertaker 1 ....  47_Undertaker 165 y así como 100 copias por cada superstar. Tambien lo borro?

**Comment by piacabrera** [comment]
Tambien me paso lo que mencionan en la issue, Could not find a part of the path '/Users/piacabrera/Desktop/Diseño Detallado/E2/RawDeal-2/RawDeal.Tests/bin/De... para todos los test

**Comment by RodrigoToroIcarte** [comment]
> Oka, estoy siguiendo el tutorial de esa issue pero surge una duda, dentro de la carpeta de net6.0 tengo demasiados archivos ademas de data, como 47_Undertaker 1 .... 47_Undertaker 165 y así como 100 copias por cada superstar. Tambien lo borro?

No, eso está bien. Solo hay que actualizar el cards.json y poner que siempre se copie. Lo que pasa es que cuando compilas el ejecutable queda en las carpetas /bin. Ahí es donde parte tu programa. Y busca el cards.json dentro de esa carpeta. Por eso hay que poner que todo lo que está en la carpeta "data" se copie a la carpeta bin al compilar. Yo les pasé el código base con todo eso configurado, pero ahora que borraste el cards.json y pusiste uno nuevo, hay que hace que también ese archivo se copie al bin. Para eso solo hay que hacer click derecho en ese archivo, ver su configuración y poner la opción de "copy always".

Una forma más artesanal de lograr lo mismo es que copies en nuevo cards.json en las carpetas bin directo:

- `RawDeal/RawDeal/bin/Debug/net6.0/`
- `RawDeal/RawDeal.Tests/bin/Debug/net6.0/`

Esas dos carpetas contienen la versión antigua del cards.json. Si copias y pegas el nuevo cards.json en ambas carpetas las cosas deberían funcionar también.

**Comment by RodrigoToroIcarte** [comment]
> Tambien me paso lo que mencionan en la issue, Could not find a part of the path '/Users/piacabrera/Desktop/Diseño Detallado/E2/RawDeal-2/RawDeal.Tests/bin/De... para todos los test

Tienes que volver a compilar. En el proyecto, click derecho y "Build solution"

**Comment by piacabrera** [comment]
<img width="1379" alt="Captura de Pantalla 2023-10-24 a la(s) 16 55 49" src="https://github.com/RodrigoToroIcarte/IIC2113-2023-1/assets/80771839/9612b0ec-da8d-4a45-9e3b-bffa3b1ec076">

Esto significa que se compilo? porque no me corre tampoco después de eso :(

** Al final copiando data a mano si funciono**

**Comment by RodrigoToroIcarte** [comment]
Sí, compiló. Pero creo que solo compilaste el proyecto RawDeal. Falta compilar el RawDeal.Tests. Si quieres compilar el proyecto completo debes hacer el Build Solution sobre el proyecto entero (que sale seleccionado):

![Screenshot from 2023-10-24 17-05-54](https://github.com/RodrigoToroIcarte/IIC2113-2023-1/assets/7000518/33c5099d-9071-4446-b814-19fb8cbc7f16)

**Comment by conicorreag** [comment]
Hola! Seguí todos los pasos, pero me sigue apareciendo el efecto antiguo, que más puedo hacer?
